### PR TITLE
Bump version to 2.1.0dev0

### PR DESCRIPTION
**Pull request by kaxil**
We are past 2.0.0 :)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by ashb — APPROVED** [review]

**Review by ashb — DISMISSED** [review]
On master branch this should be 2.1.0...

File: setup.py
Comment (kaxil) [inline setup.py]:
```suggestion
version = '2.1.0.dev0'
```

**Comment by kaxil** [comment]
> On master branch this should be 2.1.0...

Updated

**Review by XD-DENG — APPROVED** [review]

**Comment by potiuk** [comment]
Question: @kaxil - how are we going to proceed with the master changes to be released in 2.0.1 - my impression is that so far (and I thought until 2.0.1) everything in master will be part of 2.0.1 and then we 'actually' branch off and start cherry-picking. So far this is what we've done (and I think that was @ashb approach that we limit the need of cherry-picking to v2_0* branches.

Is the assumption (with this change) that we will be now manually cherry-picking the the master changes to v2_0_test * branch as we did for v1_10_test and that we should mark all the PRs that are worth cherry-picking with 2.0.1 milestone? 

I would love to clarify it :). I have a number of changes in master that I might have missed marking with '2.0.1' milestone over the last couple of days and they for sure should be part of 2.0.1

**Comment by kaxil** [comment]
> Question: @kaxil - how are we going to proceed with the master changes to be released in 2.0.1 - my impression is that so far (and I thought until 2.0.1) everything in master will be part of 2.0.1 and then we 'actually' branch off and start cherry-picking. So far this is what we've done (and I think that was @ashb approach that we limit the need of cherry-picking to v2_0* branches.
> 
> Is the assumption (with this change) that we will be now manually cherry-picking the the master changes to v2_0_test * branch as we did for v1_10_test and that we should mark all the PRs that are worth cherry-picking with 2.0.1 milestone?
> 
> I would love to clarify it :). I have a number of changes in master that I might have missed marking with '2.0.1' milestone over the last couple of days and they for sure should be part of 2.0.1

That assumption still stands correct :) -- We will rebase v2-0-test on top of Master but just remove this commit. 

At any point from now to future, we should **_try hard_** not diverge from Master and even if we need to -- try and stay as close as possible.

**Comment by ashb** [comment]
Now we've released 2.0.0 only critical/major fixes will be in a 2.0.x - most other changes can wait to 2.1.0

**Comment by ashb** [comment]
> That assumption still stands correct :) -- We will rebase v2-0-test on top of Master but just remove this commit.

I'm not sure that is true anymore. If like us to minimize cherry picking by just limiting what we put in to the 2.0 point release(s)

**Comment by kaxil** [comment]
> > That assumption still stands correct :) -- We will rebase v2-0-test on top of Master but just remove this commit.
> 
> I'm not sure that is true anymore. If like us to minimize cherry picking by just limiting what we put in to the 2.0 point release(s)

Fair point, I will need to look at what all the changes have been to the Master, my gut feeling says they would mostly be doc changes, bug fixes, breeze / CI but will need to check. If that is the case, a rebase (ofcourse still checking the contents would be same as cherry-picking).

Our goals are aligned though that we want to follow strict SemVer and not release new features and/or breaking changes in patch versions to have a similar fate as 1.10.x

**Comment by potiuk** [comment]
> Now we've released 2.0.0 we have in my mind already branched off, and only critical/major fixes will be in a 2.0.x - most other changes can wait to 2.1.0

I think it's just a matter of technicalities when it comes to branching but I agree with the principle. The only question is whether we should only cherry pick "critical fixes" and what "critical fixes" mean. 

I treat 2.0.1 mostly as "cleanup and stabilising" release. There are (and still will be some more) issues raised by first-time users, and while none of them is critical, I think we already reached a critical mass of "bugfix" release (including some of the issues that still need diagnosing and fixes) and we have not added any "feature" yet.  So for me pretty much every bugfix we have in 2.0 now an most of the "dev cleanup" should go there as well.

So technically I think it will be easier to (make sure there is no new feature added) and move the branches and take "everything" in master to 2.0.1 (and refrain from merging any new features before we release 2.0.1). That will be IMHO better than cherry-picking all those changes needed because there will be quite a lot of them. And all the docs/dev tools changes should also go 2.0.1 I believe.

Of course it also matters when we release 2.0.1 but looking at the number of issues and type of them, I'd say targetting 2.0.1 in the week 4-8 Jan might be a good idea. In which case just moving the branches might be much easier.

**Comment by potiuk** [comment]
@kaxil @ashb https://github.com/apache/airflow/pull/13394#pullrequestreview-560480362 is I think the first I see that is clearly targeted to 2.1. I think we should decide how we proceed before merging it

**Comment by ashb** [comment]
I'll be back working from Monday

**Pull request merged**
